### PR TITLE
Prayer

### PR DESCRIPTION
**Pull request by coolbop32**
### Description:

-   _Add Prayer Skill

### Changes:

-   Add the +bury and +offer commands
-   Add the ability to create spirit shields

-   [x] I have tested all my changes thoroughly.

### CODE REVIEW

File: src/commands/Minion/bury.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, boneName = '']: [null | number | string, string]) {
+		if (!msg.author.hasMinion) {
+			throw `You dont have a minion`;
+		}
+		// default bury speed
+		let speedmod = 1;
```
Comment (Alexsuperfly) [inline src/commands/Minion/bury.ts:29]: move this block below the minion busy

File: src/commands/Minion/bury.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+					deathCounter++;
+				}
+			}
+			for (let i = 0; i < deathCounter; i++) {
+				bonesLost += rand(1, 27);
+			}
+		}
+
+		// Check the user has the required bones to bury.
+		const hasRequiredLogs = await msg.author.hasItem(bone.inputBones, quantity);
```
Comment (Alexsuperfly) [inline src/commands/Minion/bury.ts:96]: logs -> bones

File: src/commands/Minion/bury.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+			if (!amountOfBonesOwned || amountOfBonesOwned === 0) throw `You have no ${bone.name}.`;
+			quantity = Math.min(
+				Math.floor(msg.author.maxTripLength / timeToBuryABone),
+				amountOfBonesOwned
+			);
+		}
+		let bonesLost = 0;
+		// See if at chaos altar
+		let chaos = false;
+		if (msg.flagArgs.chaos) {
```
Comment (Alexsuperfly) [inline src/commands/Minion/bury.ts:79]: do these calcs in the activity

File: src/commands/Minion/bury.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import { CommandStore, KlasaMessage } from 'klasa';
```
Comment (gc) [inline src/commands/Minion/bury.ts:1]: I think we need to rename bury cmd to bones, or make bury cmd only for burying, just for naming

File: src/commands/Minion/bury.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+				bone => bone.name
+			).join(', ')}.`;
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Prayer) < bone.level) {
+			throw `${msg.author.minionName} needs ${bone.level} Prayer to bury ${bone.name}.`;
+		}
+
+		// Time to bury a bone
+
```
Comment (gc) [inline src/commands/Minion/bury.ts:64]: remove this empty line

File: src/commands/Minion/bury.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+			channelID: msg.channel.id,
+			quantity,
+			duration,
+			chaos,
+			type: Activity.Prayer,
+			id: rand(1, 10_000_000),
+			finishDate: Date.now() + 30000
+		};
+
+		// Remove the bones from their bank.
```
Comment (gc) [inline src/commands/Minion/bury.ts:105]: no need to comment this

File: src/commands/Minion/bury.ts [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+		await msg.author.removeItemFromBank(bone.inputBones, quantity);
+
+		await addSubTaskToActivityTask(this.client, Tasks.SkillingTicker, data);
+
+		msg.author.incrementMinionDailyDuration(duration);
+		return msg.send(
```
Comment (gc) [inline src/commands/Minion/bury.ts:111]:
```
await addSubTaskToActivityTask(this.client, Tasks.SkillingTicker, data);
msg.author.incrementMinionDailyDuration(duration);

return msg.send(```

format like this please, so related things are together

File: src/lib/skilling/skills/prayer.ts [hunk 8]
```diff
@@ -0,0 +1,135 @@
+import { Bone, SkillsEnum } from '../types';
+import itemID from '../../util/itemID';
+import { Emoji } from '../../constants';
+
+const bones: Bone[] = [
+	{
+		name: 'bones',
```
Comment (gc) [inline src/lib/skilling/skills/prayer.ts:7]: why all bones arent cased properly? rs items always have the first letter capitalized

File: src/lib/skilling/skills/prayer.ts [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+		level: 1,
+		xp: 4.5,
+		inputBones: itemID('Bones')
```
Comment (gc) [inline src/lib/skilling/skills/prayer.ts:10]: lets rename `inputBones` to `id` to more reflect what it is, an id

File: src/lib/createables/misc.ts [hunk 10]
```diff
@@ -0,0 +1,98 @@
+import itemID from '../util/itemID';
+import { Bank } from '../types';
+
+interface Createable {
+	name: string;
+	outputItems: Bank;
+	inputItems: Bank;
+	smithingLevel?: number;
+	addOutputToCollectionLog?: boolean;
```
Comment (gc) [inline src/lib/createables/misc.ts:9]: i think it would be best if you reverted your refactors to creatables, and just add these items like normal for now, even if the file is very big, i think it should be refactored ina  different PR, nothing unrelated to prayer, unless its needed, should be in it.

File: src/tasks/minions/PrayerActivity/offeringActivity.ts [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+import Prayer from '../../../lib/skilling/skills/prayer';
+import { channelIsSendable } from '../../../lib/util/channelIsSendable';
+import { SkillsEnum } from '../../../lib/skilling/types';
+
+export default class extends Task {
+	async run({ boneID, quantity, userID, channelID }: OfferingActivityTaskOptions) {
+		const user = await this.client.users.fetch(userID);
+		const currentLevel = user.skillLevel(SkillsEnum.Prayer);
+
+		const Bury = Prayer.Bones.find(Bury => Bury.inputId === boneID);
```
Comment (gc) [inline src/tasks/minions/PrayerActivity/offeringActivity.ts:16]: why is this called `Bury`? it should be `bones`, the casing might be uppercased in others but we should start doing it properly camelcased.

File: src/tasks/minions/PrayerActivity/offeringActivity.ts [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+		const xpmod = 3.5;
```
Comment (gc) [inline src/tasks/minions/PrayerActivity/offeringActivity.ts:18]: variables should be camelcased, please check other variables too to ensure. 
```
const XPMod = 
const maxPK =
```

File: src/commands/Minion/bury.ts [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	async run(msg: KlasaMessage, [quantity, boneName = '']: [null | number | string, string]) {
+		if (!msg.author.hasMinion) {
+			throw `You dont have a minion`;
+		}
+
+		if (msg.author.minionIsBusy) {
+			return msg.send(msg.author.minionStatus);
+		}
```
Comment (gc) [inline src/commands/Minion/bury.ts:30]: delete these if's entirely, and replace them with the 2 new decorators, see here for an example: https://github.com/gc/oldschoolbot/blob/bce8983035be343963722be2328f107ab2f8bca1/src/commands/Minion/groupkill.ts#L57

and do this for anywhere else which does these 2 checks.

File: src/commands/Minion/bury.ts [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+			throw `That's not a valid bone to bury. Valid bones are ${Prayer.Bones.map(
+				bone => bone.name
+			).join(', ')}.`;
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Prayer) < bone.level) {
+			throw `${msg.author.minionName} needs ${bone.level} Prayer to bury ${bone.name}.`;
+		}
+
+		// Time to bury a bone
```
Comment (gc) [inline src/commands/Minion/bury.ts:55]: i dont think we need a comment saying `timeToBuryABone` variable is the `time to bury a bone` 😄

File: src/commands/Minion/bury.ts [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+	async run(msg: KlasaMessage, [quantity, boneName = '']: [null | number | string, string]) {
+		if (!msg.author.hasMinion) {
+			throw `You dont have a minion`;
+		}
+
+		if (msg.author.minionIsBusy) {
+			return msg.send(msg.author.minionStatus);
+		}
+		// default bury speed
+		const speedmod = 1;
```
Comment (gc) [inline src/commands/Minion/bury.ts:32]: camelcase
